The test `LlamaIntegrationTest::test_conversion` test is failing

The following command
```bash
RUN_SLOW=1 python3 -m pytest -v tests/models/llama/test_tokenization_llama.py::LlamaIntegrationTest::test_conversion
```
gives 
```bash
>           self.assertEqual(old_serialized, new_serialized)
E           AssertionError: '{\n [1465 chars]   "Sequence": {\n          "id": "B",\n      [1794589 chars]}\n}' != '{\n [1465 chars]   "SpecialToken": {\n          "id": "<s>",\n[1794837 chars]}\n}'

tests/models/llama/test_tokenization_llama.py:337: AssertionError
```


### Who can help?

@ArthurZucker

--- a/src/transformers/convert_slow_tokenizer.py
+++ b/src/transformers/convert_slow_tokenizer.py
@@ -1175,7 +1175,11 @@ class LlamaConverter(SpmConverter):
             single = f"{(bos+':0 ') * add_bos}$A:0{(' '+eos+':0') * add_eos}"
             pair = f"{single}{(' '+bos+':1') * add_bos} $B:1{(' '+eos+':1') * add_eos}"
 
-            special_tokens = [(bos, bos_token_id), (eos, eos_token_id)]
+            special_tokens = []
+            if add_bos:
+                special_tokens.append((bos, bos_token_id))
+            if add_eos:
+                special_tokens.append((eos, eos_token_id))
             return processors.TemplateProcessing(single=single, pair=pair, special_tokens=special_tokens)
 
         else:

